### PR TITLE
Feature 5 - Create recipe

### PR DESCRIPTION
**Pull request by leticiadaluzp**
#### Board:
https://www.notion.so/5-Create-Recipe-fffb29625ca681469265fa81f1780413?pvs=4 
---
#### Description:
Create a new recipe by providing the following attributes: `name` (string), `description` (text), and `ingredients` (text), `user` (User).
---
#### Tasks:
- [x] Implement the model to create a recipe and save the data to the database.
- [x] Ensure that the `name`, `description`, and `ingredients` attributes are correctly handled in the `recipes` table.
- [x] Implement RecipesController create action and display a message for success and error scenarios.
- [x] Write tests to verify that a recipe can be successfully created and saved with the correct attributes.---

### CODE REVIEW

File: app/models/recipe.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Indexes
+#
+#  index_recipes_on_user_id  (user_id)
+#
+class Recipe < ApplicationRecord
+  belongs_to :user
+
+  validates :name, presence: true
+  validates :description, presence: true
+  validates :ingredients, presence: true
```
Comment (lauperalti) [inline app/models/recipe.rb]:
```suggestion
  validates :name, :description, :ingredients, presence: true
```

File: spec/factories/recipes.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Indexes
+#
+#  index_recipes_on_user_id  (user_id)
+#
+FactoryBot.define do
+  factory :recipe do
+    name { Faker::Food.dish }
+    description { Faker::Food.description }
+    ingredients { Faker::Food.ingredient }
```
Comment (lauperalti) [inline spec/factories/recipes.rb:23]: Nice use of Faker 👏

File: spec/models/recipe_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+RSpec.describe Recipe do
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:description) }
+    it { is_expected.to validate_presence_of(:ingredients) }
+  end
+
+  describe 'associations' do
+    it { is_expected.to belong_to(:user) }
+  end
```
Comment (lauperalti) [inline spec/models/recipe_spec.rb:30]: This is great 👏

File: app/models/user.rb [hunk 4]
```diff
@@ -50,6 +50,7 @@ class User < ApplicationRecord
   include DeviseTokenAuth::Concerns::User
 
   has_many :preferences, dependent: :destroy
+  has_many :recipes, dependent: :destroy
```
Comment (lauperalti) [inline app/models/user.rb:53]: Maybe we could add a test on model user_spec for this association

File: spec/requests/recipes/recipe_spec.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    subject { post recipes_path, params: }
+
+    let(:params) { { recipe: attributes_for(:recipe) } }
+
+    context 'when logged in' do
+      let!(:user) { create(:user) }
+
+      before { sign_in user }
+
+      context 'when success' do
```
Comment (lauperalti) [inline spec/requests/recipes/recipe_spec.rb:37]: Could we add some context and tests for when the params are invalid?

File: spec/requests/recipes/recipe_spec.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        it 'creates the recipe' do
+          expect { subject }.to change(Recipe, :count).by(1)
+        end
+
+        it 'redirect to index' do
```
Comment (lauperalti) [inline spec/requests/recipes/recipe_spec.rb]: Could we test also the success message text?

**Review by lauperalti — COMMENTED** [review]
Looks great! 💪 
Left some small comments

**Comment by leticiadaluzp** [comment]
The suggested improvements were added, let me know if you have any comments

**Review by lauperalti — APPROVED** [review]
Excellent work 🚀

**Pull request merged**